server/webapp/WEB-INF/rails.new/app/controllers/api_v1/elastic/profiles_controller.rb

```diff
@@ -0,0 +1,87 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  module Elastic
+    class ProfilesController < ::ApiV1::BaseController
+      before_action :check_for_stale_request, only: [:update]
+      before_filter :check_admin_user_and_401
+
+      def index
+        profiles = elastic_profile_service.allProfiles.values.to_a
+        render DEFAULT_FORMAT => ProfilesRepresenter.new(profiles.to_a).to_hash(url_builder: self)
+      end
+
+      def show
+        profile = load_profile
+        if stale?(etag: etag_for(profile))
+          render DEFAULT_FORMAT => ProfileRepresenter.new(profile).to_hash(url_builder: self)
+        end
+      end
+
+      def update
+        profile = load_profile
+        profile_from_request = ProfileRepresenter.new(ElasticProfile.new).from_hash(params[:profile])
+
+        result = HttpLocalizedOperationResult.new
+        elastic_profile_service.update(current_user, etag_for(profile), profile_from_request, result)
+        handle_create_or_update_response(result, profile)
+      end
+
+      def create
+        result = HttpLocalizedOperationResult.new
+        profile = ProfileRepresenter.new(ElasticProfile.new).from_hash(params[:profile])
```

Comment:
Do we provide `:profile` param as a part of create operation, if not `params[:profile]` becomes null and can't create `ProfileRepresenter`. Using https://github.com/ketan/api.go.cd/blob/5dda72801af7eda946f974de05356a631c6e3053/source/includes/elastic_profiles/_30-create.md.erb as a reference to make a request.